drake/core/test/testVector.cpp

```diff
@@ -13,15 +26,17 @@ int main(int argc, char* argv[])
   state.theta = 0.2;
   state.thetadot = .3;
 
+  assert(size(state)==2);
```

Comment:
We should be careful with the usage of assert. I think that currently most of our CI builds use a release configuration, in which case this assert does nothing.